src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -281,7 +281,11 @@ trait Namers extends MethodSynthesis {
       }
       tree.symbol match {
         case NoSymbol => try dispatch() catch typeErrorHandler(tree, this.context)
-        case sym      => enterExistingSym(sym, tree)
+        case sym      =>
+          tree match {
+            case tree@Import(_, _) => enterExistingSym(sym, tree).make(tree)
```

Comment:
I think `tree.symbol` is null, at least for imports, at this point in the batch compiler, which goes to the code just above in `dispatch()`.

```
case tree @ Import(_, _)                           => enterImport(tree); returnContext = context.make(tree)
```

I'll test that theory out by adding some assertions before I merge, won't have time for a few days though.




case tree @ Import(_, _)                           => enterImport(tree); returnContext = context.make(tree)
 